assets/src/js/admin/onboarding-wizard/app/store/reducer.js

```diff
@@ -0,0 +1,18 @@
+// Import utilities
+import { getWindowData } from '../../utils';
+
+export const reducer = ( state, action ) => {
+	switch ( action.type ) {
+		case 'GO_TO_STEP':
+			if ( action.payload.step <= state.lastStep ) {
+				return {
+					...state,
+					currentStep: action.payload.step,
+				};
+			}
+			window.location.href = getWindowData( 'setupUrl' );
```

Comment:
Should the redirect be its own action? Or is it better to couple the logic within the GO_TO_STEP action?

This isn't a blocker, just something that I'm thinking through.

Reply:
This is more architectural, so I think it is worth a discussion.

Reply:
Yeah, was thinking through a similar idea. Thinking about the problem more, I realize that redirecting to the setup page has nothing to do with the store, and should be aware of the store itself. So I decoupled this logic from the go to step action, and now import it as a utility, which is fired in the ContinueButton component (where the logic to check if it is the last step or not now lives). Ref: https://github.com/impress-org/givewp/pull/4942/commits/3e660960e29365b82cede3d4de170346e69c339e